ComPact/Builders/V3/MessageBuilder.cs

```diff
@@ -106,11 +146,38 @@ private void CheckMessageHandler<T>(Action<T> messageHandler)
             {
                 throw new ArgumentNullException(nameof(messageHandler));
             }
+            CheckMessageContents();
+            _isVerified = false;
+        }
+
+        private void CheckMessageHandler<T>(Func<T,Task> messageHandler)
```

Comment:
I think it would also work with a single `private void CheckMessageHandler(object messageHandler)` replacing the two variations.

Reply:
Changed to use implicit CheckMessageHandler(object messageHandler)